flake-modules/dev/launch-test.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+: "${NIXVIM_NIX_COMMAND:=nix}"
+if [[ -z $NIXVIM_SYSTEM ]]; then
+	NIXVIM_SYSTEM=$(nix eval --raw --impure --expr "builtins.currentSystem")
+fi
+
+help() {
+	cat <<EOF
+Usage: tests [-i/--interactive] [-h/--help] [-t/test <test name>]
+
+Options:
+	-i/--interactive: Pick interactively the tests
+	-h/--help: Display this help message and exit
+	-t/--test <test name>: Only launch the specified test (conflicts with -i)
+EOF
+}
+
+if ! OPTS=$(getopt -o "hit:" -l "help,interactive,test:" -- "$@"); then
+	echo "Invalid options" >&2
+	help
+	exit 1
+fi
+
+eval set -- "$OPTS"
+
+test_name=
+interactive=false
+
+while true; do
+	case "$1" in
+	-h | --help)
+		help
+		exit 0
+		;;
+	-i | --interactive)
+		interactive=true
+		shift
+		;;
+	-t | --test)
+		test_name=$2
+		shift 2
+		;;
+	*) break ;;
+	esac
+done
+
+if [[ $interactive = true && -n $test_name ]]; then
+	echo "Can't specify both --interactive and --test" >&2
+	exit 1
+fi
+
+test_list() {
+	nix flake show --json 2>/dev/null |
+		jq '.checks."x86_64-linux" | keys | map(select(startswith("test-")))[]' --raw-output
+}
+
+if [[ $interactive = false && -z $test_name ]]; then
+	test_list |
+		xargs -I{} sh -xc "${NIXVIM_NIX_COMMAND} build .#checks.${NIXVIM_SYSTEM}.{} || exit 255" || exit $?
```

Comment:
Also when specifying multiple `installables` to `nix build` with `.#` it evaluates them separately (loads a flake multiple time). It's a 10 second for each on my VM. It should be faster something like this:

```
build -f . checks.x86_64-linux.enable-except-in-tests checks.x86_64-linux.extend checks.x86_64-linux.extra-args-tests checks.x86_64-linux.extra-files checks.x86_64-linux.home-manager-module checks.x86_64-linux.lib-tests checks.x86_64-linux.maintainers checks.x86_64-linux.man-docs checks.x86_64-linux.nixos-module checks.x86_64-linux.no-flake
```

instead of

```
nix build .#checks.x86_64-linux.enable-except-in-tests .#checks.x86_64-linux.extend .#checks.x86_64-linux.extra-args-tests .#checks.x86_64-linux.extra-files .#checks.x86_64-linux.home-manager-module .#checks.x86_64-linux.lib-tests .#checks.x86_64-linux.maintainers .#checks.x86_64-linux.man-docs .#checks.x86_64-linux.nixos-module .#checks.x86_64-linux.no-flake
```

because in the first case it evaluates only one time, then it takes attrs of it. In the second case every `.#` evaluates a flake separately.

Reply:
Ah, I see that xargs here won't combine it to one command, so `nix build` is called serially. Maybe it can be ran in batches (need to find a balance between memory consumption and speed). Example:

```bash
test_list | xargs -n 10 sh -xc "nix build -f . \"\$@\" || exit 255" --
```

but prefix `checks.${NIXVIM_SYSTEM}` should be added in jq.

Reply:
@traxys is this resolved?

Reply:
Yup